### PR TITLE
Added entity selection changes to webhook

### PR DESCRIPTION
**Pull request by darth-coder00**
### Describe your changes :

I worked on the AddWebhook for entity selection changes

#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :
<p align="center">
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [ x I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@shahsank3t @Sachin-chaurasiya

### CODE REVIEW

**Review by Sachin-chaurasiya — APPROVED** [review]

File: openmetadata-ui/src/main/resources/ui/src/components/AddWebhook/AddWebhook.interface.ts [hunk 1]
```diff
@@ -18,7 +18,11 @@ import { Webhook } from '../../generated/entity/events/webhook';
 export interface AddWebhookProps {
   data?: Webhook;
   header: string;
+  mode: 'add' | 'edit';
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/AddWebhook/AddWebhook.interface.ts:21]: Here, please use an enum.

File: openmetadata-ui/src/main/resources/ui/src/components/AddWebhook/AddWebhook.tsx [hunk 2]
```diff
@@ -166,19 +180,31 @@ const AddWebhook: FunctionComponent<AddWebhookProps> = ({
     return retVal;
   };
 
+  const getHiddenEntitiesList = (entities: Array<string> = []) => {
+    if (entities.includes('*')) {
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/AddWebhook/AddWebhook.tsx:184]: `const allEntities="*"`
Store this value in constant and use that constant here and in other places.

File: openmetadata-ui/src/main/resources/ui/src/components/common/webhook-data-card/WebhookDataCard.tsx [hunk 3]
```diff
@@ -12,67 +12,43 @@
...
   name,
   description,
   endpoint,
   status = Status.NotStarted,
-  onDelete,
-  onEdit,
+  onClick,
 }: Props) => {
+  const handleLinkClick = () => {
+    onClick && onClick(name);
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/common/webhook-data-card/WebhookDataCard.tsx:35]:
```suggestion
    onClick?.(name);
```

File: openmetadata-ui/src/main/resources/ui/src/components/dropdown/CheckBoxDropDownList.tsx [hunk 4]
```diff
@@ -37,18 +38,21 @@ const CheckBoxDropDownList = ({
...
-                className="tw-cursor-pointer"
+                className="tw-cursor-pointer tw-py-1"
                 key={index}
                 onClick={(e) => onSelect && onSelect(e, item.value as string)}>
                 <input
                   checked={Boolean(
                     selectedItems?.includes(item.value as string)
                   )}
                   className="tw-ml-3 tw-mr-2 tw-align-middle custom-checkbox"
+                  disabled={Boolean(
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/dropdown/CheckBoxDropDownList.tsx:53]:
```suggestion
                  disabled={disabledItems?.includes(item.value as string)}
```

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]
Require a few changes.

File: openmetadata-ui/src/main/resources/ui/src/components/dropdown/CheckBoxDropDownList.tsx (hunk 4 above)
Comment (darth-coder00) [reply openmetadata-ui/src/main/resources/ui/src/components/dropdown/CheckBoxDropDownList.tsx:53]: `disabledItems` can be `undefined`, that will set disabled to `undefined`

**Pull request merged**